### ISSUE TITLE
Python: Option<&ForeignClass> doesn't work

### ISSUE THREAD

**Issue by tasn** [Dushistov/flapigen-rs#339]
Passing `&ForeignClass` to a function works just fine, but passing `Option<&ForeignClass>` doesn't.
`ForiegnClass` is a newly defined class.

Errors:
`creates a temporary which is freed while still in use`
and ` borrowed value does not live long enough`.

I'd provide more a more detailed error though unfortunately #337 makes it very difficult to debug it myself.

**Comment by tasn** [comment]
Same with `Vec<&ForeignClass>`.

**Comment by tasn** [comment]
@AdamRzepka if you have any idea that would be awesome! I managed to workaround all of the other Python backend limitations, and this is the only one left!

Thanks!

**Comment by AdamRzepka** [comment]
I'll take a look at it soon. I'm almost sure, we use types like this as function arguments in our code.

**Comment by tasn** [comment]
Thanks a lot! I ended up using an ugly workaround to get this to "work". We still have a lot of copies in our API due to this, but at least it works.
https://github.com/etesync/etesync-rs/blob/etebase/python/src/glue.rs.in#L450

**Comment by AdamRzepka** [comment]
> I'm almost sure, we use types like this as function arguments in our code.

Well, it turns out, that I was wrong. We don't use function arguments like `Option<&T>`. We only use function return types like this.

The problem is, if the arguments needs a chain of conversions, the python backend just stacks all of them recursively in the single expression. This can indeed cause problems with intermediate values storage and RAII types, like `MutexLock`.

What we should do instead, is to chain the conversions and bind the intermediate values to variables, that live throughout the function call. That's how it's done in other backends.

In other words, we need to change this:

```rust
let result = function_call(
    match arg { 
        Some (inner) => Some ( ( & * super :: py_test_class :: rust_instance ( & inner , py ) . lock ( ) . unwrap ( ) ) ) ,
        None => None , 
    }
)
```

to something like this:

```rust
let arg_1 = arg_0.as_ref().map(|inner| super :: py_test_class :: rust_instance ( & arg, py ) . lock ( ) . unwrap ( ));
let arg_2 = arg_1.as_ref().map(|inner| &* inner);
let result = function_call(arg_2);
```

It shouldn't be very hard to do. Feel free to ask further questions, if you're interested in fixing this,

**Comment by tasn** [comment]
Yup, this is exactly the issue I was seeing, and I bet the temp variables will indeed fix it.
Btw, it's cool that it does the conversions recursively!
